src/apps/material/material.tsx

```diff
@@ -104,25 +106,32 @@ const Material: React.FC<MaterialProps> = ({ wid }) => {
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [data]);
 
-  // This useEffect selects the current manifestation.
   useEffect(() => {
     if (!data?.work) return;
     const { work } = data as { work: Work };
-    const type = getUrlQueryParam("type");
+
+    // If the urlType is present in manifestations.all, we will use that type,
+    // otherwise fallback to the best material type for the work.
+    const urlType = getUrlQueryParam("type");
+    const urlTypeIsPresentInManifestations =
+      filterManifestationsByType(
+        urlType as ManifestationMaterialType,
+        work.manifestations.all
+      ).length > 0;
+
     const manifestationsByMaterialType = divideManifestationsByMaterialType(
       work.manifestations.all
     );
-    // If there is no type in the url, we select one.
-    if (!type) {
-      const bestMaterialType = getBestMaterialTypeForWork(work);
-      setSelectedManifestations(manifestationsByMaterialType[bestMaterialType]);
-      setQueryParametersInUrl({
-        type: bestMaterialType
-      });
+
+    if (urlType && urlTypeIsPresentInManifestations) {
+      setSelectedManifestations(manifestationsByMaterialType[urlType]);
```

Comment:
Consider refactoring this code.

As I read it we split `work.manifestations.all` by type twice:

1. When running `filterManifestationsByType()` to determine `urlTypeIsPresentInManifestations`
2. When running `divideManifestationsByMaterialType()` to determine `manifestationsByMaterialType`

I am not sure how computationally expensive is but consider if we can do it once. One way would be to determine `urlTypeIsPresentInManifestations` from `manifestationsByMaterialType`.

Reply:
Updated the code to only do one check 👍